zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -246,17 +239,20 @@ impl<F: Field> MPTConfig<F> {
             .try_into()
             .unwrap();
 
-        let memory_columns = (0..5).map(|_| meta.advice_column()).collect::<Vec<_>>();
-
         let mut state_machine = StateMachineConfig::construct(meta);
         let mut rlp_item = MainRLPGadget::default();
-
-        let mut memory = Memory::new(memory_columns);
-        memory.allocate(meta, key_memory(false), MptCellType::MemKeyCTable);
-        memory.allocate(meta, key_memory(true), MptCellType::MemKeySTable);
-        memory.allocate(meta, parent_memory(false), MptCellType::MemParentCTable);
-        memory.allocate(meta, parent_memory(true), MptCellType::MemParentSTable);
-        memory.allocate(meta, main_memory(), MptCellType::MemMainTable);
+        let memory = Memory::new(
+            meta,
+            vec![
+                (MptCellType::MemKeyC, 3),
+                (MptCellType::MemKeyS, 3),
+                (MptCellType::MemParentC, 3),
+                (MptCellType::MemParentS, 3),
+                (MptCellType::MemMain, 3),
+            ],
+            0,
+            50,
```

Comment:
`1` hopefully okay?

Reply:
Nope 😅 the minimum that work is `5`
Cuz this is should be the total amount of queries happening in each branch, namely `max(is_start.max(R,W), is_branch.max(R,W) is_account.max(R,W), is_storage.max(R,W))`.

Reply:
Hmm should still be `1` right otherwise the current code wouldn't work? The rw_counter is per memory type and each memory is only loaded/stored at most once I think. I think you may not take into account the regions? Because the rw_counter should be reset for each reason so the cells can overlap.

Reply:
Ahhha, I figured out why:
```
// inside store function
cb.store_expression(
            name.as_str(),
            compressed_expr.expr(),
            C::default(),
            Some(self.query_write()),
        );
        self.table_conditions.push((cb.region_id, condition));
```
So the `region_id` is configured and it is taken cared of by the CB. However the `self.query_write()` is not under CB's management, that's why the whole system work but `rw_records` columns still are not reused.

Reply:
I quickly tried `1` now and it still gives an out of bounds error in `query_write`. I think it still doesn't take into account regions but I only quickly looked at it so may have missed it. In any case, I think `1` should work?

Reply:
Does my reply last time make sense? not sure if i've missed something:
 
> ```
> // inside store function
> cb.store_expression(
>             name.as_str(),
>             compressed_expr.expr(),
>             C::default(),
>             Some(self.query_write()),
>         );
>         self.table_conditions.push((cb.region_id, condition));
> ```
> 
> So the `region_id` is configured and it is taken cared of by the CB. However the `self.query_write()` is not under CB's management, that's why the whole system work but `rw_records` columns still are not reused.
But after I move back the rw columns into CellManager the rw cells should be under unified management and reused so then `1` should work.

Reply:
Okay sounds good! Unified management!